From 1dfdb05e8ffb91b79482247672eed405de7b6e9d Mon Sep 17 00:00:00 2001
From: Bogdan Marinescu <bogdan.a.marinescu@intel.com>
Date: Wed, 15 Feb 2012 16:05:58 +0200
Subject: Enable RELRO in executables and shared libraries

This patch enables generation of RELRO specific structures in ELF
files (for dynamically linked executables and shared libraries) by
adding the corresponding flags to the linker (-z relro -z now).

Issue: ANDROID-1267
Change-Id: Iff7e2f7e5f31a3fd867b023a2d8e6ade06479ba1
Signed-off-by: Bogdan Marinescu <bogdan.a.marinescu@intel.com>
---
 core/combo/TARGET_linux-x86.mk |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/core/combo/TARGET_linux-x86.mk b/core/combo/TARGET_linux-x86.mk
index 04531b2..eacbba9 100644
--- a/core/combo/TARGET_linux-x86.mk
+++ b/core/combo/TARGET_linux-x86.mk
@@ -180,6 +180,7 @@ $(hide) $(PRIVATE_CXX) \
 	$(PRIVATE_TARGET_GLOBAL_LDFLAGS) \
 	 -nostdlib -Wl,-soname,$(notdir $@) \
 	 -shared -Bsymbolic \
+	 -Wl,-z,relro,-z,now \
 	$(TARGET_GLOBAL_CFLAGS) \
 	$(PRIVATE_TARGET_GLOBAL_LD_DIRS) \
 	$(if $(filter true,$(PRIVATE_NO_CRT)),,$(PRIVATE_TARGET_CRTBEGIN_SO_O)) \
@@ -203,6 +204,7 @@ $(hide) $(PRIVATE_CXX) \
 	$(PRIVATE_TARGET_GLOBAL_LDFLAGS) \
 	-nostdlib -Bdynamic \
 	-Wl,-dynamic-linker,/system/bin/linker \
+	-Wl,-z,relro,-z,now \
 	-Wl,-z,nocopyreloc \
 	-fPIE -pie \
 	$(PRIVATE_TARGET_GLOBAL_LD_DIRS) \
-- 
1.7.4.1

